### PR TITLE
Payload value types

### PR DESCRIPTION
**Pull request by harleensahni**
Please link to the issue #…

this pull request has distinct payloadValue property names for each type. It also has an example of propensity that is separate from the number type to enforce values [0,1]

### CODE REVIEW

**Comment by wizangzing** [comment]
I don't know how valuable it is to try to have a relatively  complete set of payload types now, as opposed to adding more as needs arise, but two more types that seem useful are ranges (age=40-49) and string (car="Ford").  I have seen customer requests for the string use case where they want to track the highest propensity product to target the user with, so they want to send different values that override the previous value.  I'm not sure how useful a range type would be compared with modelling it with a boolean, as we do now.  Does it make sense to add string and/or range now?

Also, for types like number, string there are a couple of possible interpretations of a sequence of realizations.  If I realize a segment with value = 5 and then later with value = 50, am I now in two audiences or does the most recent value always win?  Do we want a field that selects which interpretation applies?

File: schemas/context/segmentmembership.schema.json [hunk 1]
```diff
@@ -29,6 +29,67 @@
...
+              },
+              "required": ["xdm:payloadPropensityValue", "xdm:payloadType"]
+            },
+            {
+              "type": "object",
+              "properties": {
+                "xdm:payloadBooleanValue": {
+                  "title": "Value",
+                  "type": "boolean",
+                  "description": "The propensity"
```
Comment (wizangzing) [inline schemas/context/segmentmembership.schema.json]: the description looks like a cut and paste error - description for type boolean should probably be something like "true or false"

File: schemas/context/segmentmembership.schema.json [hunk 2]
```diff
@@ -29,6 +29,67 @@
...
           "description": "The timestamp when the assertion of segment membership was made."
         },
+        "xdm:payload" : {
+          "title" : "Payload",
+          "description": "Values that are directly related with the segment realization. This payload exists with the same validUntil as the segment realization.",
+          "oneOf": [
+            {
+              "type": "object",
+              "properties": {
+                "xdm:payloadNumberValue": {
```
Comment (harleensahni) [inline schemas/context/segmentmembership.schema.json:39]: We could support only number type first. We'll punt on the other types. 

Create an issue for polymorophic types using oneOf and const types. This against adobe/xdm repo. cc: Kevin

File: schemas/context/experienceevent.example.1.json [hunk 3]
```diff
@@ -114,5 +114,23 @@
...
+      "xdm:segmentID": {
+        "@id": "https://data.adobe.io/entities/identity/92312748749128",
+        "xdm:namespace": {
+          "xdm:code": "AAM"
+        }
+      },
+      "xdm:lastQualificationTime": "2017-09-26T15:52:25+00:00",
+      "xdm:version": "1.0",
+      "xdm:validUntil": "2017-12-26T15:52:25+00:00",
+      "xdm:status": "realized",
```
Comment (amuraru) [inline schemas/context/experienceevent.example.1.json:129]: what's a realized status?

Comment (harleensahni) [reply schemas/context/experienceevent.example.1.json:129]: @amuraru, it comes from here: https://github.com/harleensahni/xdm/pull/3/files#diff-cd4afc92ff6cef7f5482454896437093R105 

The realized value means that the visitor entered the segment (new)

File: schemas/context/segmentmembership.schema.json (hunk 1 above)
Comment (harleensahni) [reply schemas/context/segmentmembership.schema.json]: fixed -- thanks!

**Comment by harleensahni** [comment]
@wizangzing , 

> I don't know how valuable it is to try to have a relatively complete set of payload types now, as opposed to adding more as needs arise, but two more types that seem useful are ranges (age=40-49) and string (car="Ford"). I have seen customer requests for the string use case where they want to track the highest propensity product to target the user with, so they want to send different values that override the previous value. I'm not sure how useful a range type would be compared with modelling it with a boolean, as we do now. Does it make sense to add string and/or range now?

I think adding the string type now makes sense. I think we want to move away from ranges. They should just store the raw value in the segment. And if they compose another segment of this segment, set the range in an operator there. 

> Also, for types like number, string there are a couple of possible interpretations of a sequence of realizations. If I realize a segment with value = 5 and then later with value = 50, am I now in two audiences or does the most recent value always win? Do we want a field that selects which interpretation applies?

This would belong in the segment definition. I'm only trying to help define the segment membership here.

**Comment by harleensahni** [comment]
### Moved to https://github.com/adobe/xdm/pull/411

**Pull request closed without merging**